x/auth/signing/cache.go

```diff
@@ -0,0 +1,77 @@
+package signing
+
+import (
+	"sync"
+
+	lru "github.com/hashicorp/golang-lru/v2"
+)
+
+var signatureCache *Cache
+
+const (
+	TxHashLen        = 32
+	AddressStringLen = 2 + 20*2
+)
+
+var initSigCacheOnce sync.Once
+
+// NewSignatureCache initializes the signature cache.
+// signature verification is one of the most expensive parts in verification
+// by caching it we avoid needing to verify the same signature multiple times
```

Comment:
I would mention that this is only initialized once to make it clear to the reader.